File: DSApython/Algorithm/Sorting/insertionSort.py
def insertionSort(arr):
    n = len(arr)
    for i in range(1,n):
        key = arr[i]
        j = i-1
        while j>=0 and arr[j]>key:
            arr[j+1] = arr[j]
            j-=1
        arr[j+1] = key


arr = [9,8,7,6,5,4,3,2,1]
insertionSort(arr)
print(arr)

File: DSAcpp/Algorithm/Sortings/insertionSort.cpp
#include <iostream>
#include <vector>
using namespace std;

void insertionSort(vector<int> &arr){
    for(int i=1; i<arr.size(); i++){
        int key = arr[i];
        int j=i-1;
        while (j>=0 && arr[j]>key)
        {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = key;
    }
}

int main(){
    vector<int> arr = {10,9,8,7,6,5,4,3,2,1};
    insertionSort(arr);
    for(int i:arr) cout << i << ' ';
    cout << '\n';

}